client/state/checklist/hooks.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { CHECKLIST_KNOWN_TASKS } from 'calypso/state/data-layer/wpcom/checklist/index.js';
+import getSiteChecklist from 'calypso/state/selectors/get-site-checklist';
+import { requestSiteChecklistTaskUpdate } from './actions';
+
+/**
+ * Request a update for the specified task
+ *
+ * @param siteId The site ID
+ * @param taskId The task ID you want to update
+ */
+export const useRequestSiteChecklistTaskUpdate = (
+	siteId: string,
+	taskId: typeof CHECKLIST_KNOWN_TASKS[ keyof typeof CHECKLIST_KNOWN_TASKS ]
+): void => {
+	const dispatch = useDispatch();
+	const task = useSelector( ( state ) =>
+		getSiteChecklist( state, Number( siteId ) )?.tasks.find( ( { id } ) => id === taskId )
```

Comment:
This assumes that the site checklist is loaded, which it will be if the user has loaded my-home. But if the user navigates to `/themes` or `/marketing/connections/` manually ( e.g. through changing the url, or by never having visited `/home` in that page load) then the site checklist will not be defined and this will always return null.

Reply:
Oops...I don't know that. Let me think about how to deal with it 🤔

Reply:
Now, it would fetch the site checklist if the value is null.